pkg/controllers/dynakube/extension/eec/reconciler.go

```diff
@@ -70,3 +83,87 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
 
 	return r.createOrUpdateStatefulset(ctx)
 }
+
+func (r *reconciler) reconcileTlsSecret(ctx context.Context) error {
+	query := k8ssecret.Query(r.client, r.client, log)
+
+	secret, err := query.Get(ctx, types.NamespacedName{
+		Name:      consts.GetTlsSecretName(r.dk.Name),
+		Namespace: r.dk.Namespace,
+	})
+
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return err
+	}
+
+	if r.dk.Spec.Templates.ExtensionExecutionController.TlsRefName != "" {
+		if secret != nil {
+			return query.Delete(ctx, secret)
+		}
+
+		return nil
+	}
+
+	if k8serrors.IsNotFound(err) {
+		err = r.createTlsSecret(ctx)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	if secret != nil {
+		err = r.reconcileTlsSecretExpiration(ctx, secret)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *reconciler) createTlsSecret(ctx context.Context) error {
+	cert, err := certificates.New()
+	cert.Cert.DNSNames = consts.GetCertificateAltNames(r.dk.Name)
+	cert.Cert.KeyUsage = x509.KeyUsageKeyEncipherment | x509.KeyUsageDataEncipherment
+	cert.Cert.ExtKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth}
+
+	if err != nil {
+		return err
+	}
+
+	err = cert.SelfSign()
+	if err != nil {
+		return err
+	}
+
+	certPem, pkPem, err := cert.ToPEM()
+	if err != nil {
+		return err
+	}
+
+	coreLabels := k8slabels.NewCoreLabels(r.dk.Name, k8slabels.ExtensionComponentLabel)
+	secretData := map[string][]byte{consts.TLSCrtDataName: certPem, consts.TLSKeyDataName: pkPem}
+
+	secret, err := k8ssecret.Build(r.dk, consts.GetTlsSecretName(r.dk.Name), secretData, k8ssecret.SetLabels(coreLabels.BuildLabels()))
+	if err != nil {
+		return err
+	}
+
+	secret.Type = corev1.SecretTypeTLS
+
+	query := k8ssecret.Query(r.client, r.client, log)
+
+	_, err = query.CreateOrUpdate(ctx, secret)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *reconciler) reconcileTlsSecretExpiration(ctx context.Context, secret *corev1.Secret) error {
+	// WIP
+	return nil
+}
```

Comment:
you can put it just here `GetTlsSecretName` I mean this one,

Reply:
Fixed